### PR TITLE
Make the service-catalog chart setup more secure with RBAC 

### PR DESCRIPTION
**Pull request by luxas**
Take advantage of API Aggregation
Creates the necessary RBAC rules for the apiserver and controller-manager

cc @pmorie @liggitt

### CODE REVIEW

**Comment by luxas** [comment]
So this is code that has been lying in my fork of service-catalog for use in https://github.com/luxas/kubeadm-workshop

I now noticed @MHBauer's PR that is kind of similar: https://github.com/kubernetes-incubator/service-catalog/pull/936, but not everything necessarily have to be the same.

We'll just see what happens I guess...

File: charts/catalog/templates/apiregistration.yaml [hunk 1]
```diff
@@ -13,6 +13,12 @@ spec:
   service:
     namespace: {{ .Release.Namespace }}
     name: {{ template "fullname" . }}-apiserver
+  {{- if eq .Values.apiserver.tls.ca "" }}
+  insecureSkipTLSVerify: true
+  {{- else }}
   caBundle: {{ .Values.apiserver.tls.ca }}
-  priority: 100
+  {{- end }}
+  # Extension API Servers should have a priority value higher than the core API groups that have 100
```
Comment (MHBauer) [inline charts/catalog/templates/apiregistration.yaml]: Why?

Comment (liggitt) [inline charts/catalog/templates/apiregistration.yaml]: The priority field actually split to `groupPriorityMinimum` and `versionPriority` in `v1beta1`. See https://github.com/kubernetes/kubernetes/pull/46800

The ordering was also made more comprehensible (bigger numbers have higher priority). I'd recommend something like:
```
groupPriorityMinimum: 1000
versionPriority: 5
```

File: charts/catalog/README.md [hunk 2]
```diff
@@ -41,26 +42,26 @@ chart and their default values.
...
 | `apiserver.imagePullPolicy` | `imagePullPolicy` for the apiserver | `Always` |
-| `apiserver.insecure`  | Whether to expose an insecure endpoint; keep this enabled because there are some outstanding problems with the TLS-secured endpoint | `true` |
+| `apiserver.insecure`  | Whether to expose an insecure endpoint; keep this enabled because there are some outstanding problems with the TLS-secured endpoint | `false` |
 | `apiserver.tls.cert` | Base64-encoded x509 certificate | A self-signed certificate |
 | `apiserver.tls.key` | Base64-encoded private key | The private key for the certificate above |
 | `apiserver.tls.ca` | Base64-encoded CA certificate used to sign the above certificate | |
 | `apiserver.tls.requestHeaderCA` | Base64-encoded CA used to validate request-header authentication, when receiving delegated authentication from an aggregator | *none (will disable requestheader authentication)* |
-| `apiserver.service.type` | Type of service; valid values are `LoadBalancer` and `NodePort` | `NodePort` |
+| `apiserver.service.type` | Type of service; valid values are `ClusterIP` (the most secure), `LoadBalancer` and `NodePort` | `ClusterIP` |
```
Comment (MHBauer) [inline charts/catalog/README.md]: I think this change makes sense if we are going to enable this by default, as we won't have any other use for an external access of the apiserver.

File: charts/catalog/templates/apiserver-deployment.yaml [hunk 3]
```diff
@@ -1,3 +1,63 @@
+kind: ServiceAccount
```
Comment (MHBauer) [inline charts/catalog/templates/apiserver-deployment.yaml]: It probably does make more sense to shove the rbac stuff into the deployments that need it set up. 

Is there any guidance on the difference or preference for multi-yaml split by `---` or using a    `List` object?

Comment (luxas) [inline charts/catalog/templates/apiserver-deployment.yaml]: I prefer `---` personally at least

File: charts/catalog/README.md (hunk 2 above)
Comment (luxas) [inline charts/catalog/README.md]: Yes, NodePort shouldn't be the default in an AA world

**Comment by luxas** [comment]
> If you're an expert on RBAC, I would have preferred your help in creating and reviewing #936 rather than making a new PR.

I can definitely help out on that PR. As said, I wrote this code quite long ago, but first now realized I should upload a PR of it and didn't see yours.

File: charts/catalog/templates/apiregistration.yaml (hunk 1 above)
Comment (luxas) [inline charts/catalog/templates/apiregistration.yaml]: Thanks, I hadn't seen that it was updated between alpha and beta

**Comment by arschles** [comment]
This will need a rebase

**Comment by pmorie** [comment]
@luxas I think this can be closed now that #936 is in, correct?

**Comment by kibbles-n-bytes** [comment]
@luxas bump

**Comment by luxas** [comment]
@kibbles-n-bytes ah, forgot about this one. I'll check what's left to do here, update and let you know.
Thanks for the ping

**Comment by pmorie** [comment]
Closing - I believe this is obsolete.

**Pull request closed without merging**